core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -117,4 +120,39 @@ private static void deleteFiles(FileIO io, Set<ManifestFile> allManifests) {
           }
         });
   }
+
+  /**
+   * Load a custom {@link FileIO} implementation.
+   * The implementation must have a no-arg constructor.
+   * If the class implements {@link org.apache.hadoop.conf.Configurable}, we will also set the Hadoop configuration.
```

Comment:
Javadoc paragraphs require `<p>` between them or else the docs are all one paragraph. Also, it is more clear and direct not to refer to the library maintainers with "we", like this: `If the class implements {@link Configurable}, a Hadoop config will be passed using {@link Configurable#setConf(Configuration)}.`